Rubberduck.Refactorings/Rename/RenameRefactoring.cs

```diff
@@ -13,6 +13,8 @@
 using Rubberduck.VBEditor.ComManagement;
 using Rubberduck.VBEditor.SafeComWrappers;
 using Rubberduck.VBEditor.Utility;
+using NLog;
+using Rubberduck.Resources;
```

Comment:
Is this used anywhere?

Reply:
`using NLog `is there due to a logged warning message when there is a parser suspension failure.  The logged warning was present in parser suspension code of `ExtractInterface` which I used as the model for the `RenameRefactoring` changes.  So - `using NLog` is occurs in both.

`using Rubberduck.Resources` is not used (I'll clean that up on my next sync).

Reply:
Regarding earlier comment 

> the rename refactoring is used quite commonly so it'll be a good test for how well suspension works out in real world before we make all refactoring adhere to it

The PR suspends the parser for renaming Modules and Projects only - which I'm suspecting is used less frequently than other renaming actions.  Is there interest in modifying the logic to suspend the parser for all rename refactoring types?

Reply:
I have to say that I do not agree that refactorings should always work under parser suspension. For this PR, having it only for the module and project renames is fine.

Reply:
The idea is to prevent VBIDE events from triggering a reparse before the refactoring is ready to trigger it; renaming a project or module normally triggers a reparse through custom events, so suspending shuts down that response and lets the refactoring do its job (alternatively we could stop firing these events). This isn't needed in other rename scenarios.

Reply:
I did not doubt that it makes sense for these scenarios. My doubts is about generally suspending in refactorings, even if the refactoring does not cause a separatebreparse itself.

Reply:
The thing is that by definition, refactoring means rewriting, which means a change in the code module and therefore triggers a reparse?

I am having trouble imagining what a refactoring that didn't require parse would look like.

Reply:
If the refactoring does not cause separate reparses via VBE events, we already guarantee that the state is consistent via the rewrite session.

Reply:
So, if I was to summarize the above: Any refactoring that renames or creates a module or project should suspend the parser during the refactoring.  So, only the `Rename `refactoring and `ExtractInterface `refactoring meet this criteria and contain parser suspension code.